File: app/src/main/java/com/didar/sample/MyAdapter.java
package com.didar.sample;

import android.support.annotation.NonNull;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.didar.mylibrary.Callback;
import com.didar.extendedrecyclerview.ExtendedRecyclerViewAdapter;


import java.util.List;

public class MyAdapter extends ExtendedRecyclerViewAdapter {
    List<Integer> list;

    public MyAdapter(int layout, List<Integer> list, Callback callback) {
        super(layout, list, callback);
        this.list = list;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
        View view = viewHolder.itemView;

        TextView textView = view.findViewById(R.id.textView);
        textView.setText(list.get(i).toString());
    }
}

File: app/src/main/java/com/didar/sample/MyAdapter.kt
package com.didar.sample

import com.didar.extendedrecyclerview.Callback
import com.didar.extendedrecyclerview.ExtendedRecyclerViewAdapter
import kotlinx.android.synthetic.main.row_list.view.*

class MyAdapter(layout: Int, internal var list: List<Int>, callback: Callback) :
        ExtendedRecyclerViewAdapter(layout, list, callback) {

    override fun onBindViewHolder(viewHolder: ExtendedRecyclerViewAdapter.ViewHolder, i: Int) {
        viewHolder.itemView.textView.text = list[i].toString()
    }
}
